src/transport.c

```diff
@@ -108,10 +108,13 @@ static int find_address(unsigned char ipaddr[BOOTH_IPADDR_LEN],
 				break;
 
 		if (matched == node->addrlen) {
-			*address_bits_matched = matched * 8;
-			*me = node;
-			did_match = EXACT_MATCH;
-			break;
+			if((matched_tmp < matched * 8)||((node->type == SITE)&&(matched_tmp == matched * 8)))
```

Comment:
On Thu, Aug 25, 2016 at 06:20:44AM -0700, Jan Pokorný wrote:
> > @@ -108,10 +108,13 @@ static int find_address(unsigned char ipaddr[BOOTH_IPADDR_LEN],
> >  				break;
> >  
> >  		if (matched == node->addrlen) {
> > -			*address_bits_matched = matched * 8;
> > -			*me = node;
> > -			did_match = EXACT_MATCH;
> > -			break;
> > +			if((matched_tmp < matched * 8)||((node->type == SITE)&&(matched_tmp == matched * 8)))
> 
> I've also observed "same subnet" vs. fuzzy matching issues.

Before or after this commit:

b2e06e84180c845d44c12a209075524a2063da93

Namely, some cloud testers complained about the issue, but after
this it worked for them (and for me).